csharp/src/Google.Protobuf/UnknownField.cs

```diff
@@ -73,8 +73,7 @@ public override bool Equals(object other)
             {
                 return true;
             }
-            UnknownField otherField = other as UnknownField;
-            return otherField != null
+            return other is UnknownField otherField
```

Comment:
Are you happy to outdent the lines below so that they're indented by 4 spaces instead of 7? Don't worry if that's fiddly for some reason.